code/obj/item.dm

```diff
@@ -739,6 +735,7 @@
 	return null
 
 /obj/item/proc/attack_self(mob/user)
+	SEND_SIGNAL(src, COMSIG_ITEM_ATTACK_SELF, user)
```

Comment:
Again there are possible parent-calling problems here etc. but it's probably fine for now.